pulsar-broker/src/main/java/org/apache/pulsar/PulsarClusterMetadataSetup.java

```diff
@@ -370,7 +373,7 @@ static void createPartitionedTopic(MetadataStore configStore, TopicName topicNam
             PartitionedTopicMetadata existsMeta = getResult.get();
 
             // Only update metadata if the partitions should be modified
-            if (existsMeta.partitions < numPartitions) {
+            if (existsMeta.partitions < numPartitions && force) {
```

Comment:
Seems `force` should not check the partition numbers, update directly...

Reply:
I don't quite understand what you mean 🤔

Reply:
if (force) {
  ....
} else if (existsMeta.partitions < numPartitions) {
 ...
}

Reply:
My idea, wait for others.

Reply:
We do not support reducing the number of TC.